frontend/ui/widget/datetimewidget.lua

```diff
@@ -101,67 +132,182 @@ function DateTimeWidget:init()
-        date_month_hour = month_hour_widget,
-        date_year = year_widget,
+        value_hold_step = self.day_hold_step or 3,
+    }
+    self:mergeLayoutInHorizontal(day_widget)
+
+    hour_widget = NumberPickerWidget:new{
+        show_parent = self,
+        value = self.hour,
+        value_min = self.hour_min or 0,
+        value_max = self.hour_max or 23,
+        value_step = 1,
+        value_hold_step = self.hour_hold_step or 4,
+    }
+    self:mergeLayoutInHorizontal(hour_widget)
+    min_widget = NumberPickerWidget:new{
+        show_parent = self,
+        value = self.min,
+        value_min = self.min_min or 0,
+        value_max = self.min_max or 59,
+        value_step = 1,
+        value_hold_step = self.min_hold_step or 10,
+    }
+    self:mergeLayoutInHorizontal(min_widget)
+    sec_widget = NumberPickerWidget:new{
+        show_parent = self,
+        value = self.sec,
+        value_min = self.sec_min or 0,
+        value_max = self.sec_max or 59,
+        value_step = 1,
+        value_hold_step = self.sec_hold_step or 10,
+    }
+    self:mergeLayoutInHorizontal(sec_widget)
+
+    local separator_date = TextBoxWidget:new{
+        text = "–",
+        alignment = "center",
+        face = self.title_face,
+        bold = true,
+        width = math.floor(math.min(self.screen_width, self.screen_height) * 0.02),
     }
-    self:mergeLayoutInHorizontal(day_min_widget)
-    local separator_space = TextBoxWidget:new{
-        text = self.is_date and "–" or ":",
+    local separator_time = TextBoxWidget:new{
+        text = _(":"),
         alignment = "center",
         face = self.title_face,
         bold = true,
-        width = math.floor(math.min(self.screen_width, self.screen_height) *
-            (self.is_date and 0.02 or 0.05)),
+        width = math.floor(math.min(self.screen_width, self.screen_height) * 0.05),
+    }
+    local separator_date_time = TextBoxWidget:new{
+        text =  _("/"),
+        alignment = "center",
+        face = self.title_face,
+        bold = true,
+        width = math.floor(math.min(self.screen_width, self.screen_height) * 0.02),
     }
     local date_group = HorizontalGroup:new{
             align = "center",
-            year_widget,
-            separator_space,
-            month_hour_widget,
-            separator_space,
-            day_min_widget,
+            year_widget, -- 1
+            separator_date, -- 2
+            month_widget, -- 3
+            separator_date, -- 4
+            day_widget, -- 5
+            separator_date_time, -- 6
+            hour_widget, -- 7
+            separator_time, -- 8
+            min_widget, -- 9
+            separator_time, -- 10
+            sec_widget, -- 11
         }
-    if not self.is_date then
+
+    -- remove unused number pickers
+    if not self.sec then
+        table.remove(date_group, 11)
+        table.remove(date_group, 10)
```

Comment:
Feels a bit odd to create them all, and then remove the not-needed - but may be it's easier that way.
Anyway, check if the widgets you build and remove have a `:free()`, and if yes, call it once removed.
Or may be we just don't care, it's rarely used, and gc should take care of it (in theory), dunno.

But it seems you may still access them all below - so may be don't :free() them here - and this should better be done in :onClose(). Dunno :)

Reply:
Thanks, I will simplify that (and the focusmanager quirk)